### ISSUE TITLE
[Bug]: LogSender dependency cannot be found in React Native projects

### ISSUE THREAD

**Issue by BanDroid** [AndroidIDEOfficial/AndroidIDE#1274]
### What happened?

react native project have their own gradle configuration, the configuration seems compatible with AndroidIDE, only error when build in the app because of logsender, build in terminal is working just fine.

here's the [logs](https://t.me/androidide_discussions/63782/90665).
here's the [project](https://t.me/androidide_discussions/63782/90672) in case you want to see the configuration.

### What's the expected behavior?

build success

### What version of AndroidIDE you're using?

v2.5.2 (debug builds)

### Relevant log output

```shell
-
```

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by itsaky** [comment]
Sure, I'll look into it.

**Comment by BanDroid** [comment]
@itsaky maybe add a fallback when logsender isn't enabled? because i tried to disable logsender in developer settings, it still trying to install logsender dependencies in the project

**Comment by itsaky** [comment]
With the above commit, the log sender dependency won't be added if log sender is disabled in AndroidIDE's developers options.

**Comment by BanDroid** [comment]
it's still trying to add logsender dependency, i have disabled logsender and also ide logs show the logsender.isEnabled to false, but somehow the build failed with the same error.

```
ToolingApiServer          D   ToolingApiMain             Arguments from tooling client:  [--init-script, /data/data/com.itsaky.androidide/files/home/.androidide/init/init.gradle, -Pandroid.aapt2FromMavenOverride=/data/data/com.itsaky.androidide/files/home/.androidide/aapt2, -Pandroidide.logsender.isEnabled=false, --stacktrace, --build-cache]  
```

**Comment by itsaky** [comment]
Check the build output, does it say `LogSender is disabed`?

**Comment by BanDroid** [comment]
> Check the build output, does it say `LogSender is disabled`?

no, [here's](https://t.me/androidide_discussions/63735/90733) the new build logs, it still the same but with logsender disabled in developer settings

**Comment by BanDroid** [comment]
`init.gradle` contains this
```
initscript {
    repositories {
        
        // Always specify the snapshots repository first
        maven {
            // Add snapshots repository for AndroidIDE CI builds
            url "https://s01.oss.sonatype.org/content/repositories/snapshots/"
        }
        
        mavenCentral()
        google()
    }

    dependencies {
        classpath 'com.itsaky.androidide:gradle-plugin:2.5.3-beta-42ac69785-SNAPSHOT'
    }
}

apply plugin: com.itsaky.androidide.gradle.AndroidIDEInitScriptPlugin
```

what happen if i remove manually the logsender plugin in here?

**Comment by itsaky** [comment]
> what happen if i remove manually the logsender plugin in here?

I think this would fix the problem for you (but it's just a workaround).

Currently, the init script is only used to add LogSender to the project. So, even if you edit the init script to not apply the plugins, the project initialisation would work just fine in the IDE (you won't be able to use LogSender though).

**Comment by BanDroid** [comment]
> If you want to use this workaround, just remove this line :
> > apply plugin: com.itsaky.androidide.gradle.AndroidIDEInitScriptPlugin

i would just comment that line instead of removing it, to be safe.

**Comment by BanDroid** [comment]
in the new latest debug version, everytime syncing will regenerate init.gradle file, and now the build or sync always failed because cannot resolve the repository

**Comment by MrIkso** [comment]
@itsaky initscript.gradle
   ```
      initscript {
        repositories {
          maven {
            mavenCentral()
            
            // Add snapshots repository for AndroidIDE CI builds
            url "https://s01.oss.sonatype.org/content/repositories/snapshots/"
          }
        }
    
        dependencies {
          classpath '${BuildConfig.packageName}:gradle-plugin:${publishingVersion}'
        }
      }
      
      gradle.settingsEvaluated { settings ->
        settings.dependencyResolutionManagement.repositories {
          // For release builds
          mavenCentral()
          
          // For AndroidIDE CI builds
          maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
        }
      }
      
      gradle.projectsLoaded {
        rootProject.subprojects.forEach {sub ->
          sub.afterEvaluate {
            sub.apply plugin: com.itsaky.androidide.gradle.AndroidIDEGradlePlugin
          }
        }
      }
      
    ```


Your old code successfully worked for new/old syntax plugins

**Comment by itsaky** [comment]
Fixed in 2775702.
